util/abs/provision_pe_xl_nodes.rb

```diff
@@ -164,30 +319,58 @@ def check_nodes_yaml(file)
 #   create_params_json(hosts, output_dir)
 def create_params_json(hosts, output_dir)
   params_json = PARAMS_JSON
+  output_path = "#{File.expand_path(output_dir)}/params.json"
+
+  puts "Replacing parameters in params.json: "
 
   # replace parameters for each host
   hosts.each do |host|
     hostname = host[:hostname]
     role = host[:role]
     param = "$#{role.upcase}$"
 
-    puts "Replacing parameters: "
-    puts " hostname: #{hostname}"
-    puts " role: #{role}"
-    puts " parameter: #{param}"
-    puts
+    puts " hostname: #{hostname}, role: #{role}, parameter: #{param}"
 
     # replace parameters
     params_json = params_json.gsub(param, hostname)
   end
 
-  File.write("#{output_dir}/params.json", params_json)
+  puts
+  puts "Writing #{output_path}"
+  puts
+
+  File.write(output_path, params_json)
+
+  check_params_json(output_path) if TEST
 end
 
-abs_id = ARGV[0] || ABS_ID
-output_dir = ARGV[1] || "./"
+# Checks the params.json file to ensure it has been written correctly
+#
+# @author Bill Claytor
+#
+# @param [String] file The 'params.json' file to check
+#
+# @example
+#   check_params_json(file)
+def check_params_json(file)
+  puts "Checking #{file}..."
+  puts
+
+  contents = File.read(file)
+  puts contents
+
+  puts "Parsing JSON..."
+  puts
+
+  json = JSON.parse contents
+  puts json
+  puts
+
+  puts "Verifying JSON..."
+  install = json["install"]
```

Comment:
How is this a validation? If the JSON contents are an empty hash '{}', the results is:
```
Verified parameter 'install':
```
I would expect this function to ensure that the `install` key exists, that it's value is not empty, and possibly return a boolean.  Otherwise I have to know to really keep an eye on the output below to to the validation by eye manually.

Reply:
Great point, thanks! I updated `check_params_json` and `check_nodes_yaml` to raise an error if the expected parameter does not match the expected value.